x-pack/plugin/esql/qa/testFixtures/src/main/resources/row.csv-spec

```diff
@@ -317,3 +317,10 @@ warning:java.lang.NumberFormatException: For input string: \"two\"
 strings:keyword |ints:integer
 [one, two]      |null
 ;
+
+arraysWithNegatives
+row integers = [1, -2, -300, -2147483648], longs = [1, -2, -2147483649], longs2 = [1, -2, -9223372036854775808], doubles = [1, -.455, -2.43, 3.4, -0.12, 8];
```

Comment:
In this test do also add some whitespaces between the sign and the actual number.
Also, add some `+` as well: `[+ 112, -      234.0, + .3, -   .5]`

Also, I would argue that parentheses should be allowed as well: `-(-1)`. Have a look at SQL implementation, especially `ExpressionTests.testNegativeLiteral`.

Reply:
> In this test do also add some whitespaces between the sign and the actual number.
Also, add some + as well: [+ 112, -      234.0, + .3, -   .5]

👍 

> Also, I would argue that parentheses should be allowed as well: -(-1). Have a look at SQL implementation, especially ExpressionTests.testNegativeLiteral.

Not sure this is what we want in this context. Parentheses are for expressions, here we are parsing plain constants.
I guess (correct me if I'm wrong) the original rationale was to validate types at parse time, so that multi-values are consistent in type. If we change this now, we'll have to add a set of additional validation and conversion rules at planning time for this.
To me it seems more an enhancement than a fix, so something for a separate PR probably, maybe with lower priority.
There is something more to do in this context anyway, eg. support square brackets in EVAL as well.

Reply:
I agree this is something for a different/future PR, but for completeness sake I think parentheses can be used with numeric literals. Also, I don't think the validation and parsing of such literals interferes with what happens at planning time. `-(-1)` should properly be parsed and the Parser should emit a `Literal(1, INTEGER)` just like it would do for a simple `1` or `+1`.

Reply:
@astefan I just pushed a fix that supports parentheses and multiple unary signs